src/parseMessageDefinition.js

```diff
@@ -190,20 +193,28 @@ export function parseMessageDefinition(messageDefinition: string) {
     .map((line) => line.trim())
     .filter((line) => line);
 
-  let definitionLines: string[] = [];
+  let definitionLines: { isJson: boolean, line: string }[] = [];
   const types: RosMsgDefinition[] = [];
+  let nextDefinitionIsJson: boolean = false;
   // group lines into individual definitions
   allLines.forEach((line) => {
-    // skip comment lines
-    if (line.indexOf("#") === 0) {
+    // ignore comment lines unless they start with #pragma rosbag_parse_json
+    if (line.indexOf("#") === 0 && line.indexOf("#pragma rosbag_parse_json") !== 0) {
       return;
     }
+
     // definitions are split by equal signs
     if (line.indexOf("==") === 0) {
+      nextDefinitionIsJson = false;
       types.push(buildType(definitionLines));
       definitionLines = [];
     } else {
-      definitionLines.push(line);
+      definitionLines.push({ isJson: nextDefinitionIsJson, line });
+      nextDefinitionIsJson = false;
+    }
+
+    if (line.indexOf("#pragma rosbag_parse_json") === 0) {
```

Comment:
#nit could pull this into a bool to avoid doing it twice `line.indexOf("#pragma rosbag_parse_json")`